### PR TITLE
Update the Ubuntu OS loader install steps

### PR DESCRIPTION
**Pull request by jgutbub**
These steps use Ubuntu 20.04 and walk the
reader through the following:

-terminal output when booting the USB installer
-creating a GPT on the destination media
-creating the individual partitions on the
 destination media (FAT32, root, swap)
-the general installation process
-downloaing the container generation tools
-creating a container from the installed files
-boot option configuration to boot the container

Fix a minor issue in EHL documenation too.

Signed-off-by: James Gutbub <james.gutbub@intel.com>

### CODE REVIEW

**Comment by jgutbub** [comment]
> Thank you for the great steps! :-) BTW, I also tried to install Ubuntu using OsLoader. Follow the normal installation without creating customized partition layout. It turned out to be working as well. That might be a even simpler way for user.

Below is what I am getting when I install using OS Loader and no additional steps. What should the boot info be to access the EXT4 partition? The installation setup an MBR where partition 1 is FAT, partition 2 is an extended partition, and partition 5 is the EXT4 rootfs:

image

**Comment by mauricema** [comment]
> > Thank you for the great steps! :-) BTW, I also tried to install Ubuntu using OsLoader. Follow the normal installation without creating customized partition layout. It turned out to be working as well. That might be a even simpler way for user.
> 
> Below is what I am getting when I install using OS Loader and no additional steps. What should the boot info be to access the EXT4 partition? The installation setup an MBR where partition 1 is FAT, partition 2 is an extended partition, and partition 5 is the EXT4 rootfs:
> 
> image

SBL currently does not support extended partition.  In your case, I guess it won't be able to boot. In my case, I remember it did not use extended partition.  Maybe it depends on the size of the storage device.  Anyway, direct install does not work, we just follow your original method.   Thank you for trying it on your side.

File: source/how-tos/boot-ubuntu.rst [hunk 1]
```diff
@@ -5,85 +5,203 @@ Boot Ubuntu
...
+The following steps have been verified with Ubuntu Linux 20.04 LTS (64-bit).
 
 .. important:: Traditional Linux booting using vmlinuz, initrd, and config.cfg is only supported with the debug build of |SPN| or if you have **self.HAVE_VERIFIED_BOOT = 0** in your BoardConfig.py; you can test using debug build initially but for release build support consider packaging the vmlinuz, initrd, and config.cfg into container image format. See :ref:`create-container-boot-image` for more details.
 
 **STEP 1:** Build |SPN|
 
 .. note:: The BoardConfig.py option **self.ENABLE_GRUB_CONFIG = 1** should be set when building |SPN|.
 
+.. note:: If you encounter any error loading the vmlinuz or initrd files due to space allocation issues you may need to increase the value of your **self.PLD_HEAP_SIZE** set in your board's BoardConfig.py and re-build |SPN| and flash as a new IFWI.
+
```
[the remark this answers is not in the document]
Comment (jgutbub) [reply source/how-tos/boot-ubuntu.rst:19]: Added.

File: source/how-tos/boot-ubuntu.rst [hunk 2]
```diff
@@ -5,85 +5,203 @@ Boot Ubuntu
...
 **STEP 2:** Flash IFWI image to the board
 
-**STEP 3:** Download Ubuntu Desktop 18.04 ISO image from Ubuntu website
+**STEP 3:** Download Ubuntu Desktop 20.04 ISO image from Ubuntu website
 
 **STEP 4:** Create a USB flash drive from the ISO image with Rufus tool
```
[the remark this answers is not in the document]
Comment (jgutbub) [reply source/how-tos/boot-ubuntu.rst]: Added screenshots.

File: source/how-tos/boot-ubuntu.rst [hunk 3]
```diff
@@ -5,85 +5,203 @@ Boot Ubuntu
...
-7. Remove USB flash drive and reboot
+Proceed to click the `+` button to add the next partition.
 
-|SPN| should boot to Ubuntu automatically from eMMC now.
+.. image:: /images/ubuntu-steps/ubuntu_step_12.png
+         :width: 600
+         :alt: Add swap partition
 
-.. note:: If you encounter any error loading the vmlinuz or initrd files due to space allocation issues you may need to increase the value of your **self.PLD_HEAP_SIZE** set in your board's BoardConfig.py and re-build |SPN| and flash as a new IFWI.
+This partition is optional for hibernation support: add a swap partition.
```
[the remark this answers is not in the document]
Comment (jgutbub) [reply source/how-tos/boot-ubuntu.rst]: Removed unnecessary info.

File: source/how-tos/boot-ubuntu.rst [hunk 4]
```diff
@@ -5,85 +5,203 @@ Boot Ubuntu
...
+         :alt: Continue testing select
+
+9. Go back into a terminal window and double check the media devices with the `lsblk` command making note of the media Ubuntu was just installed to. Proceed to run `mkdir root` and mount partition 2 of the newly installed media to this folder, in the case of this example we run `sudo mount /dev/nvme1n1p2 ./root/`. Check the contents of `./root/boot` and note the contents, there are 2 files of interest, a kernel image and a corresponding initrd.
+
+.. image:: /images/ubuntu-steps/ubuntu_step_19.png
+         :alt: Mount installation media root partition
+
+.. note:: Your kernel and initrd versions may be different from the ones displayed in the above image, e.g. 5.11.0-27-generic.
+
+10. If you have a network connection you can download the tools utilized to create a signed OS container for booting Ubuntu (required for release mode boot of |SPN|) by running the commands below. Alternatively you can download these scripts onto a USB and copy them into the Ubuntu environment manually or run the next steps on a different system altogether (you will need to copy the kernel image and initrd files to the other system however).
```
[the remark this answers is not in the document]
Comment (jgutbub) [reply source/how-tos/boot-ubuntu.rst:178]: Added a note and designated as an additional optional section.

File: source/how-tos/boot-ubuntu.rst [hunk 5]
```diff
@@ -5,85 +5,203 @@ Boot Ubuntu
...
+13. Shutdown the Ubuntu system, remove the USB flash drive, and reboot the system.
+
+|SPN| should now be able to boot the Ubuntu sbl_os container file.
+
+If you are facing difficulty check the `boot` command output to confirm your boot options are configured similar to the below (depending on the media you installed Ubuntu to)::
+
+  Idx|ImgType|DevType|DevNum|Flags|HwPart|FsType|SwPart|File/Lbaoffset
+    0|      0|  NVME |    0 |   0 |    0 | EXT2 |    1 | /boot/sbl_os *Current
+
+This is indicating that NVMe device number instance 0 (if you have more than 1 NVMe you may need to try the other device instances) should have a 2nd partition (e.g. SwPart=1) with an EXT2/3/4 file system and the file `/boot/sbl_os` which is the Ubuntu container image generated in the earlier steps. If earlier you skipped adding the FAT32 partition your SwPart value should be set to 0 instead.
```
[the remark this answers is not in the document]
Comment (jgutbub) [reply source/how-tos/boot-ubuntu.rst:226]: Added a note to the top of the page about this.

**Review by mauricema — APPROVED** [review]
Looks good.

**Pull request merged**